lib/jekyll/page.rb

```diff
@@ -123,12 +123,11 @@ def process(name)
     # layouts      - The Hash of {"name" => "layout"}.
     # site_payload - The site payload Hash.
     #
-    # Returns nothing.
+    # Returns String rendered page.
     def render(layouts, site_payload)
-      site_payload["page"] = to_liquid
-      site_payload["paginator"] = pager.to_liquid
-
-      do_layout(site_payload, layouts)
+      renderer = Renderer.new(site, self, site_payload)
```

Comment:
This class still includes `Convertible`. You should use the `_renderer` method here instead and just set payload and layouts. Like:

```ruby
self.output = _renderer.tap do |renderer|
  renderer.payload = site_payload
  renderer.layouts = layouts
end.run
@_renderer = nil
self.output
```

:)